zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlParser.java

```diff
@@ -217,25 +224,78 @@ private boolean parseSource(HttpMessage message, Source source, int depth, Strin
+                        HTMLElementName.P,
+                        HTMLElementName.TITLE,
+                        HTMLElementName.H1,
+                        HTMLElementName.H2,
+                        HTMLElementName.H3,
+                        HTMLElementName.H4,
+                        HTMLElementName.H5,
+                        HTMLElementName.H6,
+                        HTMLElementName.LI,
+                        HTMLElementName.BLOCKQUOTE);
+
+        for (String tag : elementsWithText) {
+            elements = source.getAllElements(tag);
+            for (Element el : elements) {
+                Source s = new Source(el.getContent());
+                if (!parseSource(message, s, depth, baseURL)) {
+                    Matcher matcher = PLAIN_COMMENTS_URL_PATTERN.matcher(s.toString());
+                    while (matcher.find()) {
+                        String foundMatch = matcher.group().trim();
+                        if (baseTagSet) {
+                            if (!baseURL.endsWith("/")) {
+                                baseURL += "/";
+                            }
+                            if (('/' == foundMatch.charAt(0)) && (0 != foundMatch.indexOf("//"))) {
+                                foundMatch = foundMatch.substring(1);
+                            }
+                        }
+                        processURL(message, depth, foundMatch, baseURL);
+                        resourcesfound = true;
+                    }
+                }
+            }
+        }
+
         // Process META elements
         elements = source.getAllElements(HTMLElementName.META);
         for (Element el : elements) {
             // If we have http-equiv attribute, then urls can be found.
             String equiv = el.getAttributeValue("http-equiv");
+            String name = el.getAttributeValue("name");
             String content = el.getAttributeValue("content");
             if (equiv != null && content != null) {
 
                 // For the following cases:
                 // http-equiv="refresh" content="0;URL=http://foo.bar/..."
                 // http-equiv="location" content="url=http://foo.bar/..."
-                if (equiv.equalsIgnoreCase("refresh") || equiv.equalsIgnoreCase("location")) {
+                if (equiv.equalsIgnoreCase("refresh")
+                        || equiv.equalsIgnoreCase("location")
+                        || equiv.equalsIgnoreCase("content-security-policy")) {
                     Matcher matcher = URL_PATTERN.matcher(content);
                     if (matcher.find()) {
                         String url = matcher.group(1);
                         processURL(message, depth, url, baseURL);
                         resourcesfound = true;
                     }
                 }
+            } else if (name != null && content != null) {
+                if (name.equalsIgnoreCase("msapplication-config")
+                        && !content.equalsIgnoreCase("none")) {
```

Comment:
Merge the ifs, the name null check can be dropped if used the literal in the left hand side.

Reply:
Done.